tailcall-tracker/src/tracker.rs

```diff
@@ -1,77 +1,51 @@
-use reqwest::header::{HeaderName, HeaderValue};
+use chrono::{DateTime, Utc};
+use tokio::time::Duration;
 
 use super::Result;
 use crate::check_tracking::check_tracking;
-use crate::event::Event;
+use crate::collect::ga::GaTracker;
+use crate::collect::posthog::PostHogTracker;
 
-const API_SECRET: &str = "GVaEzXFeRkCI9YBIylbEjQ";
-const MEASUREMENT_ID: &str = "G-JEP3QDWT0G";
-const BASE_URL: &str = "https://www.google-analytics.com";
+#[async_trait::async_trait]
+pub trait EventCollector: Send + Sync {
+    async fn dispatch(&self, event_name: &str, start_time: DateTime<Utc>) -> Result<()>;
+}
 
-///
-/// Base structure to track usage of the CLI application
-#[derive(Debug, Clone)]
 pub struct Tracker {
-    base_url: String,
-    api_secret: String,
-    measurement_id: String,
+    collectors: Vec<Box<dyn EventCollector>>,
     is_tracking: bool,
 }
 
 impl Default for Tracker {
     fn default() -> Self {
+        let ga_tracker = GaTracker::default();
+        let posthog_tracker = PostHogTracker::default();
         Self {
-            base_url: BASE_URL.to_string(),
-            api_secret: API_SECRET.to_string(),
-            measurement_id: MEASUREMENT_ID.to_string(),
+            collectors: vec![Box::new(ga_tracker), Box::new(posthog_tracker)],
             is_tracking: check_tracking(),
         }
```

Comment:
We can set the start time in the Tracker over here instead of passing it from outside via init and dispatch.